include/fmt/format-inl.h

```diff
@@ -423,6 +423,18 @@ class fp {
     lower.f <<= lower.e - upper.e;
     lower.e = upper.e;
   }
+
+  void compute_float_boundaries(fp& lower, fp& upper) const {
+    constexpr int min_normal_e = std::numeric_limits<float>::min_exponent -
+                                 std::numeric_limits<double>::digits;
+    significand_type half_ulp = 1 << (std::numeric_limits<double>::digits -
+                                      std::numeric_limits<float>::digits - 1);
+    if (min_normal_e > e) half_ulp <<= min_normal_e - e;
+    upper = normalize<0>(fp(f + half_ulp, e));
+    lower = fp(f - (half_ulp >> (f == implicit_bit && e > min_normal_e)), e);
```

Comment:
The condition is `e > min_normal_e`, not `e >= min_normal_e`, because the smallest positive normal float is the average of its neighbors. (The same is true for the smallest positive normal double but `compute_boundaries` does not account for that which may be either a bug or irrelevant.)

Reply:
> The same is true for the smallest positive normal double but compute_boundaries does not account for that which may be either a bug or irrelevant.

Great catch! It's definitely a bug which fortunately doesn't affect the output. Will be fixed in https://github.com/fmtlib/fmt/commit/eb879a680d996260c5268c09d6c855cf0583d7ac.